source/slang/slang-ir-insts.h

```diff
@@ -3255,6 +3255,26 @@ struct IRCastFloatToInt : IRInst
     FIDDLE(leafInst())
 };
 
+FIDDLE()
+struct IRCastStorageToLogicalBase : IRInst
+{
+    FIDDLE(baseInst())
+    IRInst* getVal() { return getOperand(0); }
+    IRInst* getBufferType() { return getOperand(1); }
+};
+
+FIDDLE()
+struct IRCastStorageToLogical : IRCastStorageToLogicalBase
+{
+    FIDDLE(leafInst())
+};
+
+FIDDLE()
+struct IRCastStorageToLogicalDeref : IRCastStorageToLogicalBase
+{
+    FIDDLE(leafInst())
+};
+
```

Comment:
These seem like the kind of IR instruction `struct`s that it should be possible to auto-generate from `slang-ir-insts.lua`. Did you try making use of the automatic generation?

Reply:
deleted `IRCastStorageToLogical` and `IRCastStorageToLogicalDeref`.

`IRCastStorageToLogicalBase` does need to exist, because fiddle cannot generate the accessors yet.